### PR TITLE
Bing image picker 2.0

### PR DESCRIPTION
**Pull request by maartenvdh**
### Motivation

New custom element similar to the Flickr image picker only in this case with the Bing API.

### Checklist

- [x] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [x] Tests are passing
- [x] Docs have been updated (if applicable)
- [x] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### How to test

If you want to use the Bing image picker in your project in Kentico Cloud, follow these steps:

* In Kentico Cloud open the Content types tab
* Create or edit a content model to which you want to add the Bing image picker
* Add the **Custom element** content element
* Configure the content element
* Use the following URL as Hosted code URL (HTTPS): https://kentico.github.io/custom-element-samples/BingImagePicker/bingimage-input-component.html
* Get API key for MS Cognitive Services: https://azure.microsoft.com/en-us/try/cognitive-services/
* Provide JSON parameters for the custom element to configure the API Key:

```json
{
    "apikey": "[YOUR API KEY]"
}
```

### CODE REVIEW

File: BingImagePicker/bingimage-input-component.html [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+                        if (fileName) {
+                            CustomElement.setValue(fileName);
+                            $("#BingImages").html($("<img>").attr("src", fileName));
+                            $("#search").val(fileName);
+                        }
+                        updateSize();
+                    });
+                })
+                .fail(function() {
+                    console.log("error");
```
Comment (Simply007) [inline BingImagePicker/bingimage-input-component.html]: Would it be possible to display an error as it is in [flickr image picker](https://github.com/Kentico/custom-element-samples/blob/master/FlickrImagePicker/flickrimage-input-component.html#L150)?

Comment (maartenvdh) [reply BingImagePicker/bingimage-input-component.html]: Done :-)

File: BingImagePicker/README.md [hunk 2]
```diff
@@ -0,0 +1,980 @@
...
+# Usage
+
+If you want to use the Bing image picker in your project in Kentico Cloud, follow these steps:
+
+* In Kentico Cloud open the Content types tab
+* Create or edit a content model to which you want to add the Bing image picker
+* Add the **Custom element** content element
+* Configure the content element
+* Use the following URL as Hosted code URL (HTTPS): https://kentico.github.io/custom-element-samples/BingImagePicker/bingimage-input-component.html
+* Get API key for MS Cognitive Services: https://azure.microsoft.com/en-us/try/cognitive-services/
```
Comment (Simply007) [inline BingImagePicker/README.md]: Could you add a note to specify which API key is required - I mean to add `Bing Search APIs v7`

File: BingImagePicker/bingimage-input-component.html [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+                    try {
+                        $.each(data.value, function (i, item) {
+                            img = `<div class='gallery'><img class='Bingimg' src='${item.contentUrl}'/>`;
+                            img += `<div class='title'>${item.name}</div></div>`;
+                            $(img).appendTo("#BingImages");
+                        });
+                    } catch (err) {
+                        // Initialization with Kentico Custom element API failed (page displayed outside of the Kentico UI)
+                        console.error(err);
+                        $("<div class='error'><strong>Images cannot be loaded</strong><br/>Check if you're API key and secret are correct.</div>").appendTo("#flickrImages");
```
Comment (Simply007) [inline BingImagePicker/bingimage-input-component.html]:
```suggestion
                        $("<div class='error'><strong>Images cannot be loaded</strong><br/>Check if you're API key and secret are correct.</div>").appendTo("#BingImages");
```

File: BingImagePicker/bingimage-input-component.html [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+                            CustomElement.setValue(fileName);
+                            $("#BingImages").html($("<img>").attr("src", fileName));
+                            $("#search").val(fileName);
+                        }
+                        updateSize();
+                    });
+                })
+                .fail(function() {
+                    console.log("error");
+                    $("<div class='error'><strong>Images cannot be loaded</strong><br/>Check if you're API key and secret are correct.</div>").appendTo("#flickrImages");
```
Comment (Simply007) [inline BingImagePicker/bingimage-input-component.html]:
```suggestion
                    $("<div class='error'><strong>Images cannot be loaded</strong><br/>Check if you're API key and secret are correct.</div>").appendTo("#BingImages");
```

**Pull request merged**